nimble/host/src/ble_store_util.c

```diff
@@ -230,16 +315,43 @@ ble_store_util_delete_oldest_peer(void)
 int
 ble_store_util_status_rr(struct ble_store_status_event *event, void *arg)
 {
+    int rc = BLE_HS_EUNKNOWN;
+    ble_addr_t peer_id_addr;
+    int num_peers;
+
     switch (event->event_code) {
     case BLE_STORE_EVENT_OVERFLOW:
         switch (event->overflow.obj_type) {
-            case BLE_STORE_OBJ_TYPE_OUR_SEC:
-            case BLE_STORE_OBJ_TYPE_PEER_SEC:
-            case BLE_STORE_OBJ_TYPE_CCCD:
-                return ble_gap_unpair_oldest_peer();
-
-            default:
-                return BLE_HS_EUNKNOWN;
+        case BLE_STORE_OBJ_TYPE_OUR_SEC:
+        case BLE_STORE_OBJ_TYPE_PEER_SEC:
+            return ble_gap_unpair_oldest_peer();
+        case BLE_STORE_OBJ_TYPE_CCCD:
+            if ((rc = ble_gap_unpair_oldest_peer()) == BLE_HS_ENOENT) {
```

Comment:
I was thinking to call here other version of `unpair_oldest` which would take as a parameter `ble_addr_t ` of device which we don't want to unpair.
If the only device which could be unpair is the one provided, then unpair is not done and this function returns error.

In such a case we know that we cannot store CCC and there is nothing we can do more here. All the other code is not needed then.

Reply:
@rymanluk That is what I have done actually in `ble_store_util_subscribed_cccds` !! What I think is `ble_gap_unpair_oldest_peer` is not anyway going to unpair current peer (the one for which we want to make space), so introducing similar function with `ble_addr_t` as input won't be really that helpful. Please let me know your thoughts on this.

The issue I believe is only confined to CCCDs, so  `ble_gap_unpair_oldest_peer()) == BLE_HS_ENOENT` will make sure that we do not have any bonded devices and still we are out of storage space, which can be used to conclude that we need to clean old CCCDs (except the current one).

Reply:
@prasad-alatkar are you sure that `ble_gap_unpair_oldest_peer()` will not unpair current peer? For me it looks it will.

Reply:
I believe @rymanluk is correct, if the current peer exceeds the cccd limit it will likely unpair that peer and disconnect. I like the do nothing if full approach here.

Reply:
@rymanluk @h2zero , disregard my earlier comment, yes current peer will be unpaired, and we should simply return error in that case.

I was thinking of adding utility function (`ble_store_clean_old_cccd`) which will delete CCCDs of all unbonded peers. This can be called first in case of `CCCD overflow` and if it fails to delete peer then we can continue with  modified `ble_gap_unpair_oldest_peer()`. Please let me know your views on it.

Reply:
I believe that `ble_store_util_delete_peer()` deletes all what we need.

Reply:
Hi @rymanluk I think for some peers `CCCD` will exist but `peer_sec` and `our_sec` won't exist (Ref: [subscribe-event ble_gatts_clt_cfg_access](https://github.com/apache/mynewt-nimble/blob/9e26fbe97b3d0bb3a22f15ed9c6035be648d1104/nimble/host/src/ble_gatts.c#L782)). So we need mechanism to clean these stored CCCDs, that is why I have introduced `ble_store_clean_old_cccd()` function.